File: paninitorunaments/src/main/java/com/paninitorunaments/paninitorunaments/service/TeamService.java
package com.paninitorunaments.paninitorunaments.service;

import com.paninitorunaments.paninitorunaments.dto.TeamRankingDto;
import com.paninitorunaments.paninitorunaments.entity.Team;

import java.util.List;

public interface TeamService {

    List<Team> getAll();

    Team getById(Long id);

    List<Team> getByEmail(String email);

    List<TeamRankingDto> getRankings();
}

File: panini-wc-ui/src/app/services/team.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root',
})
export class TeamService {
  private http = inject(HttpClient);

  private api =
    'http://localhost:9094/api/teams';

  getAll(): Observable<any[]> {
    return this.http.get<any[]>(this.api);
  }

  getTeamById(
    teamId: number
  ) {
    return this.http.get<any>(
      `${this.api}/${teamId}`
    );
  }

  getById(
    id: number
  ): Observable<any> {
    return this.http.get<any>(
      `${this.api}/${id}`
    );
  }

  getByEmail(
    email: string
  ): Observable<any[]> {
    return this.http.get<any[]>(
      `${this.api}/email/${email}`
    );
  }

  getRankings(): Observable<any[]> {
    return this.http.get<any[]>(
      `${this.api}/rankings`
    );
  }
}
